cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskManager.java

```diff
@@ -140,61 +213,84 @@ public void addBalancingProposals(Collection<BalancingProposal> proposals,
   }
 
   /**
-   * Mark the given tasks as in progress.
+   * Mark the given tasks as in progress. Tasks are executed homogeneously -- all tasks have the same balancing action.
    */
   public void markTasksInProgress(List<ExecutionTask> tasks) {
-    for (ExecutionTask task : tasks) {
-      _inProgressTasks.add(task);
-      _inProgressPartitions.add(task.proposal.topicPartition());
-      if (task.proposal.balancingAction() == BalancingAction.REPLICA_MOVEMENT) {
-        if (task.sourceBrokerId() != null) {
-          _inProgressPartMovementsByBrokerId.put(task.sourceBrokerId(),
-                                                 _inProgressPartMovementsByBrokerId.get(task.sourceBrokerId()) + 1);
-        }
-        if (task.destinationBrokerId() != null) {
-          _inProgressPartMovementsByBrokerId.put(task.destinationBrokerId(),
-                                                 _inProgressPartMovementsByBrokerId.get(task.destinationBrokerId()) + 1);
+    if (!tasks.isEmpty()) {
+      // Get the common balancing action for the given tasks.
+      BalancingAction taskBalancingAction = tasks.iterator().next().proposal.balancingAction();
+
+      for (ExecutionTask task : tasks) {
+        // Add task to the relevant task in progress.
+        BalancingAction balancingAction = task.proposal.balancingAction();
+        _executionTaskTracker.inProgressTasksFor(balancingAction).add(task);
+        _executionTaskTracker.pendingProposalsFor(balancingAction).remove(task.proposal);
+        _inProgressPartitions.add(task.proposal.topicPartition());
+        if (taskBalancingAction == BalancingAction.REPLICA_MOVEMENT) {
+          if (task.sourceBrokerId() != null) {
+            _inProgressPartMovementsByBrokerId.put(task.sourceBrokerId(),
+                                                   _inProgressPartMovementsByBrokerId.get(task.sourceBrokerId()) + 1);
+          }
+          if (task.destinationBrokerId() != null) {
+            _inProgressPartMovementsByBrokerId.put(task.destinationBrokerId(),
+                                                   _inProgressPartMovementsByBrokerId.get(task.destinationBrokerId()) + 1);
+          }
         }
       }
     }
   }
 
   /**
-   * Mark the given task as aborted.
+   * Mark the successful completion of a given task. Only normal execution may yield successful completion.
    */
-  public void markTaskAborted(ExecutionTask task) {
-    _abortedTasks.add(task);
+  public void markTaskDone(ExecutionTask task) {
```

Comment:
It is a little confusing when looking at the following three methods:
1. markTaskDone()
2. markTaskHealthiness()
3. completeTasks()

Would it be possible to merge 1 and 2 to markTaskHealthiness()?
Thinking about this a bit more, it seems a little weird to have the healthiness of NORMAL. What do you think about merging the state and healthiness to just have PENDING, IN_PROGRESS, ABORTED, DEAD and SUCCEEDED?